spec/src/main/asciidoc/chapters/language/expressions.adoc

```diff
@@ -466,40 +466,33 @@ A quantified comparison expression is satisfied if and only if:
 - the quantifier is `#any#` or `#some#` and the result list of the subquery contains at least one value which would make the equality or inequality expression evaluate to true if it occurred as the right operand in place of the subquery, or
 - the quantifier is `#all#` and every value belonging to the result list of the subquery would make the equality or inequality expression evaluate to true if it occurred as the right operand in place of the subquery.
 
-=== Ordering
+[[natural-order]]
+=== Natural order
 
-Every <<atomic-values,atomic value type>> can, in principle, be equipped with a total order. An order for a type determines the result of inequality comparisons, and the effect of the <<Order clause>>.
+Every <<atomic-values,atomic value type>> can, in principle, be equipped with a total order. An order for a type determines the result of <<equality-and-inequality,inequality comparisons>> and the effect of the <<order-clause>>.
 
-For numeric types, and for date, time, and datetime types, the total order is unique and completely determined by the semantics of the type. Jakarta Query implementations must sort these types according to their natural order, that is, the order must agree with the order defined by Java.
+For any numeric type, or for any date or datetime type, there is a _natural order_ completely determined by the semantics of the type. Jakarta Query implementations must sort these types according to their natural order:
 
-Boolean values must be ordered so that `false < true` is satisfied.
+- smaller numeric values come before larger numeric values, and
```

Comment:
There is a corner case here. Imagine the scenario where we have the same numeric value, but with different precision.

For example, with BigDecimal, **1.0** and **1.00** are numerically equal but have different representations. In Java, BigDecimal.compareTo() treats them as equal, whereas some databases may distinguish between them based on storage/scale.

The next question is: Should the Jakarta Query overwrite the database behavior? 

 Maybe it’s better to clarify that implementations should follow the database’s native numeric ordering, instead of trying to enforce one universal rule at the spec level.

Reply:
OK, so I believe what you're saying here is that relative to the definition of equality given be `BigDecimal.equals()`, the natural order of `BigDecimal` is actually a partial order rather than a total order. That's try, but we have not said anything at all about equality in this section, and in [this proposal](https://github.com/jakartaee/query/pull/63/files) I define equality on terms of the partial order, and *not* in terms of a method of the Java object.

So in fact `<` is a total order with respect to the notion of `=` in Jakarta Query.

It might be worth a footnote to clarify that Jakarta Query `=` is not the same as `BigDecimal.equals()` in Java, but I don't believe that we have anything which asserts that it is. (If we do, that's a bug.)

So I think this is fine.